### PR TITLE
fix(suite-native): Fix QR code scan doesnt progress to the next screen

### LINKED ISSUE #6371: QR code scan doesn't progress to the next screen
**Describe the bug**
Scanning a QR doesn't progress.

**Info:**
-   OS: Android
-   Version: 22.8.0 (119)

**How to reproduce**
1. Install app 
2. Toggle QR code scanning
3. Put camera in front of QR code

**Expected behaviour**
After a successful scan, the 'checking balances' screen should appear.

**Actual behaviour***
- The manual input field changes as if the field has focus
- No tapping responds

**Screenshots**
https://user-images.githubusercontent.com/216917/191753406-7ea17c0a-8c97-47ba-91a4-55a32c6c4e7d.mp4

### PR DESCRIPTION
**Pull request by bouchja1**
## Description

Scanning a QR does progress and after a successful scan, the 'checking balances' screen appears.

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/6371

## Screenshots:

https://user-images.githubusercontent.com/2227592/193222049-d442fa30-d303-4f96-87ac-b4926fa93139.mov

### CODE REVIEW

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx [hunk 1]
```diff
@@ -80,6 +81,9 @@ export const XpubScanScreen = ({
     };
 
     const handleRequestCamera = () => {
+        reset({
```
Comment (Nodonisko) [inline suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx:84]: Why is this better than setValue?

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx [hunk 2]
```diff
@@ -92,12 +96,15 @@ export const XpubScanScreen = ({
 
     const onXpubFormSubmit = handleSubmit(goToAccountImportScreen);
 
-    const handleXpubResult = (xpubAddress?: string) => {
-        if (xpubAddress) {
-            setValue('xpubAddress', xpubAddress);
-            onXpubFormSubmit();
-        }
-    };
+    const handleXpubResult = useCallback(
```
Comment (Nodonisko) [inline suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx:99]: So whole problem was just missing useCallback?

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx [hunk 3]
```diff
@@ -67,7 +67,8 @@ export const XpubScanScreen = ({
     const form = useForm<XpubFormValues>({
         validation: xpubFormValidationSchema,
     });
-    const { handleSubmit, setValue } = form;
+    const { handleSubmit, setValue, watch, reset } = form;
+    const watchXpubAddress = watch('xpubAddress');
```
Comment (Nodonisko) [inline suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx:71]: Why do we need this?

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx (hunk 1 above)
Comment (bouchja1) [reply suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx:84]: I follow this statement in the react-hook-form Q&A (https://github.com/react-hook-form/react-hook-form/discussions/7671)

- also `reset` will not only reset your form values, but it'll wipe the form's states like _isDirty, isValid_ etc. and this is what we want to do - have a new and fresh default form state.

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx (hunk 2 above)
Comment (bouchja1) [reply suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx:99]: I would say "the main" problem.

As you can see, there is `<Camera onResult={handleXpubResult} />` component for camera scanning. `handleXpubResult` sets the value for input field: `setValue('xpubAddress', xpubAddress);` and if you set the value, the whole form is updated and `handleXpubResult` passed as `onResult` prop to the component `Camera` is changed.

`onResult` is one of the array dependencies for useEffect in `Camera`... so `handleXpubResult` was called again and again and again...

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx (hunk 3 above)
Comment (bouchja1) [reply suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx:71]: This is the necessary dependency for `useCallback` mentioned in the other comment https://github.com/trezor/trezor-suite/pull/6428#discussion_r984333568. According to it: setValue changes form state, then submit onXpubFormSubmit is rerendered, it means handleXpubResult is rerendered, setValue changes state...

Comment (Nodonisko) [reply suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx:71]: I somehow doesn't understand, from what I see it looks like seems redundant. It is used in that if I think we don't need it here.

**Review by Nodonisko — APPROVED** [review]

**Pull request merged**